server/main.go

```diff
@@ -31,9 +35,17 @@ func main() {
 	errc := make(chan error)
 	go func() { errc <- serve(*port, errc) }()
 	go func() { gameLoop(errc) }()
-	select {
-	case err := <-errc:
-		log.Fatal("error somewhere", err)
+	for {
+		select {
+		case err := <-errc:
+
+			if strings.HasPrefix(err.Error(), "fatal") {
```

Comment:
i like it. maybe what we can do here is 

```go
switch err.(type){
    case fatalErr:
       log.Fatal(err)
   default:
       log.Println("err: ", err)
}
```

and then define an error type `fatalErr` which we can return

i just prefer this pattern than having to remember to put the prefix `fatal` onto a string